lib/pharos/phases/configure_etcd_changes.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+require 'json'
+
+module Pharos
+  module Phases
+    class ConfigureEtcdChanges < Pharos::Phase
+      title 'Configure etcd member changes'
+
+      def call
+        store_initial_cluster_state
+
+        if initial_cluster_state == 'existing' && etcd.healthy?
+          remove_old_members
+          add_new_members
+        end
+      end
+
+      def store_initial_cluster_state
+        return if cluster_context['etcd-initial-cluster-state']
+
+        if @ssh.file('/etc/kubernetes/manifests/pharos-etcd.yaml').exist?
+          cluster_context['etcd-initial-cluster-state'] = 'existing'
+        else
+          cluster_context['etcd-initial-cluster-state'] = 'new'
+        end
+      end
+
+      def add_new_members
+        member_list = etcd.members
+        new_members = @config.etcd_hosts.select { |h|
+          !member_list.find { |m|
+            m['name'] == peer_name(h) && m['peerURLs'] == ["https://#{h.peer_address}:2380"]
+          }
+        }
+        if new_members.size > (member_list.size / 2.0).ceil
```

Comment:
This is not perfect.. suggestions?

Reply:
`new_members.size / member_list.size.to_f >= 0.5` ?

Reply:
Yeah, the `new_members.size > (member_list.size / 2.0).ceil` would allow you to add three new members to an existing cluster of two members: `['c', 'd', 'e'].size > (['a', 'b', 'c', 'd', 'e'].size / 2.0).ceil => false`

The `new_members.size / member_list.size.to_f >= 0.5` is also wrong, it would refuse to add 1 new member bring the cluster to two members: `['a'].size / ['a', 'b'].size.to_f >= 0.5 => true`

Reply:
It's still a bit unclear if we can actually add more than one peer at once.

Reply:
However, I figure the existing condition is probably correct... if you want to be able to scale the cluster up from one member to two members, then you will have to allow breaking the quorum by one node when adding the new member.

If you wanted the cluster to retain quorum with the new members added, then it would be something like `unless member_list.size - new_members.size >= member_list.size / 2 + 1`, but that would make it impossible to scale up a cluster from one node.

Reply:
> It's still a bit unclear if we can actually add more than one peer at once.

Yes, the best approach would be to add one new peer at a time, waiting for the new peer to successfully join the cluster before adding the next one.